CONTRIBUTING.md

```diff
@@ -254,6 +255,26 @@ Introducing new [Parse Server configuration][config] parameters requires the fol
 1. Add test cases to ensure the correct parameter value validation. Parse Server throws an error at launch if an invalid value is set for any configuration parameter.
 1. Execute `npm run docs` to generate the documentation in the `/out` directory. Take a look at the documentation whether the description and formatting of the newly introduced parameters is satisfactory.
 
+## Versioning
+
+> The following versioning system is applied since Parse Server 5.0.0 and does not necessarily apply to previous releases.
+
+Parse Server follows [semantic versioning](https://semver.org) with a flavor of [calendric versioning](https://calver.org). Semantic versioning makes Parse Server easy to upgrade because breaking changes only occur in major releases. Calendric versioning gives an additional sense of how old a Parse Server release is and allows for future Long-Term Support concepts, which Parse Server currently does not provide.
+
+Syntax: `[major]`**.**`[minor]`**.**`[patch]`**-**`[pre-release-label]`
+
+- The `major` version increments every year and may include changes that are *not* backwards compatible.
+- The `minor` version increments during the year and may include new features or improvements of existing features that are backwards compatible.
+- The `patch` version increments during the year and may include bug fixes that are backwards compatible.
+- The `pre-release-label` is optional for pre-release versions such as:
+  - `-alpha` (likely to contain bugs, likely to change in features until release)
+  - `-beta` (likely to contain bugs, no change in features until release)
+  - `-rc0` (unlikely to contain bugs, no change in features until release)
+
+Exceptions:
+- The `major` version may increment during the year in the unlikely event that a breaking change is so urgent that it cannot wait for the next yearly release. This should be as rare as the planetary alignment of Jupiter and Saturn.
```

Comment:
I find the timescales here quite ambiguous...
> The `major` version increments every year and may include changes that are *not* backwards compatible.

Could be taken to mean at least once per year. But based on the other info I think you're suggesting once per year?

> The `minor` version increments during the year and may include new features or improvements of existing features that are backwards compatible.

The use of "during the year" implies to me that the major release is done at the start of each year. I'm assuming the intention is during the 12 months following the last major release?

---
I don't know if it's just me but I find it a little confusing 🤷‍♂️

Reply:
Maybe something like this :

The `major` versions are released once or twice per year depending if breaking changes need to be introduced.


---

The `minor` versions are released during the year to address non-breaking changes and may include [......]

Reply:
Or simply copy from [this](https://semver.org/#summary) :)

Reply:
@TomWFox Do you mean it should read `The major version increments once a year...` instead of `every year`?

> The use of "during the year" implies to me that the major release is done at the start of each year.  I'm assuming the intention is during the 12 months following the last major release?

Your first reading is correct, we increment at the beginning of every year. This year is special as we start with 5.0.0 mid-year, but from 2022 onward, the first release of the year will be `6.0.0`, maybe not on Jan 1, but sometime during Jan.

Reply:
There is really no need to "force" a major versions every year if there is no breaking changes. 

If breaking changes need to be introduced, schedule a release, but not the other way around.

GitLab have a very good release cycle, and its well defined in their docs, maybe you can find something interesting

https://about.gitlab.com/handbook/engineering/releases/#overview-and-terminology

Reply:
@sadortun the reasons for binding the major version to a calendar year:
- is it very likely that there will be at least 1 breaking change accumulated in 12 months (historically we can expect many more than 1), so your worry about changing the version without having a breaking change is merely theoretical
- we do not want to release a breaking version sooner than 1 year; we need to get rid of the current upgrade fatigue and get more developers on the release train and stay up-to-date with their deployment
- a 1 year cycle addresses the aforementioned aspects and lays the groundwork for an easy LTS concept, which we already identified as something we will likely try out in the future, I estimate in 2022/23.
- starting the 1 year cycle at the beginning of the year makes a release more semantic and intuitive for everyone especially in LTS context

The reason why there are so many different release concepts out there is because the best release concept is what works best for the releaser. You mention GitLab, but I could mention MongoDB that will change to a new versioning system in 2022 that is similar to the what is proposed here. Please feel free to lay out your arguments against the suggested calver concept in more detail.

Reply:
Thanks for the details ! Its much more clear now :) !

Reply:
I didn't mean to cut off your thought, we are still looking for what works best for us (which we will likely only find out by doing and adapting); if there is anything specific from GitLab that we should incorporate in the versioning system of release / branch system, please feel free to point throw in it 🙂

Reply:
Hehehehe! Dont worry, i was just finding your `one major per year` a bit much, but after your explanation it make sense !

Reply:
@TomWFox Which changes if any would you suggest?

Reply:
@TomWFox Are there any wording changes you'd suggest?

I believe once we decide for one way or another, we'll learn and improve. There may be adaptions necessary for automation / mono-repo which are hard to foresee at this stage, but this PR should give a direction where we aim to go.

Reply:
Thanks for the clarification, I think `- The major version increments at the beginning of every year and may include changes that are *not* backwards compatible.` would be more clear.

Reply:
Do you think we should say "with the first release of a / every year"? "The beginning" could be understood to be Jan 1.

Reply:
Yes, “with the first release of every year” I’d say.

Reply:
Done!